%  LaTeX support: latex@mdpi.com 
%  For support, please attach all files needed for compiling as well as the log file, and specify your operating system, LaTeX version, and LaTeX editor.

%=================================================================
\documentclass[remotesensing,article,submit,pdftex,moreauthors]{Definitions/mdpi}

%--------------------
% Class Options:
%--------------------
%----------
% journal
%----------
% Choose between the following MDPI journals:
% acoustics, actuators, addictions, admsci, adolescents, aerobiology, aerospace, agriculture, agriengineering, agrochemicals, agronomy, ai, air, algorithms, allergies, alloys, analytica, analytics, anatomia, animals, antibiotics, antibodies, antioxidants, applbiosci, appliedchem, appliedmath, applmech, applmicrobiol, applnano, applsci, aquacj, architecture, arm, arthropoda, arts, asc, asi, astronomy, atmosphere, atoms, audiolres, automation, axioms, bacteria, batteries, bdcc, behavsci, beverages, biochem, bioengineering, biologics, biology, biomass, biomechanics, biomed, biomedicines, biomedinformatics, biomimetics, biomolecules, biophysica, biosensors, biotech, birds, bloods, blsf, brainsci, breath, buildings, businesses, cancers, carbon, cardiogenetics, catalysts, cells, ceramics, challenges, chemengineering, chemistry, chemosensors, chemproc, children, chips, cimb, civileng, cleantechnol, climate, clinpract, clockssleep, cmd, coasts, coatings, colloids, colorants, commodities, compounds, computation, computers, condensedmatter, conservation, constrmater, cosmetics, covid, crops, cryptography, crystals, csmf, ctn, curroncol, cyber, dairy, data, ddc, dentistry, dermato, dermatopathology, designs, devices, diabetology, diagnostics, dietetics, digital, disabilities, diseases, diversity, dna, drones, dynamics, earth, ebj, ecologies, econometrics, economies, education, ejihpe, electricity, electrochem, electronicmat, electronics, encyclopedia, endocrines, energies, eng, engproc, entomology, entropy, environments, environsciproc, epidemiologia, epigenomes, est, fermentation, fibers, fintech, fire, fishes, fluids, foods, forecasting, forensicsci, forests, foundations, fractalfract, fuels, future, futureinternet, futurepharmacol, futurephys, futuretransp, galaxies, games, gases, gastroent, gastrointestdisord, gels, genealogy, genes, geographies, geohazards, geomatics, geosciences, geotechnics, geriatrics, grasses, gucdd, hazardousmatters, healthcare, hearts, hemato, hematolrep, heritage, higheredu, highthroughput, histories, horticulturae, hospitals, humanities, humans, hydrobiology, hydrogen, hydrology, hygiene, idr, ijerph, ijfs, ijgi, ijms, ijns, ijpb, ijtm, ijtpp, ime, immuno, informatics, information, infrastructures, inorganics, insects, instruments, inventions, iot, j, jal, jcdd, jcm, jcp, jcs, jcto, jdb, jeta, jfb, jfmk, jimaging, jintelligence, jlpea, jmmp, jmp, jmse, jne, jnt, jof, joitmc, jor, journalmedia, jox, jpm, jrfm, jsan, jtaer, jvd, jzbg, kidneydial, kinasesphosphatases, knowledge, land, languages, laws, life, liquids, literature, livers, logics, logistics, lubricants, lymphatics, machines, macromol, magnetism, magnetochemistry, make, marinedrugs, materials, materproc, mathematics, mca, measurements, medicina, medicines, medsci, membranes, merits, metabolites, metals, meteorology, methane, metrology, micro, microarrays, microbiolres, micromachines, microorganisms, microplastics, minerals, mining, modelling, molbank, molecules, mps, msf, mti, muscles, nanoenergyadv, nanomanufacturing,\gdef\@continuouspages{yes}} nanomaterials, ncrna, ndt, network, neuroglia, neurolint, neurosci, nitrogen, notspecified, %%nri, nursrep, nutraceuticals, nutrients, obesities, oceans, ohbm, onco, %oncopathology, optics, oral, organics, organoids, osteology, oxygen, parasites, parasitologia, particles, pathogens, pathophysiology, pediatrrep, pharmaceuticals, pharmaceutics, pharmacoepidemiology,\gdef\@ISSN{2813-0618}\gdef\@continuous pharmacy, philosophies, photochem, photonics, phycology, physchem, physics, physiologia, plants, plasma, platforms, pollutants, polymers, polysaccharides, poultry, powders, preprints, proceedings, processes, prosthesis, proteomes, psf, psych, psychiatryint, psychoactives, publications, quantumrep, quaternary, qubs, radiation, reactions, receptors, recycling, regeneration, religions, remotesensing, reports, reprodmed, resources, rheumato, risks, robotics, ruminants, safety, sci, scipharm, sclerosis, seeds, sensors, separations, sexes, signals, sinusitis, skins, smartcities, sna, societies, socsci, software, soilsystems, solar, solids, spectroscj, sports, standards, stats, std, stresses, surfaces, surgeries, suschem, sustainability, symmetry, synbio, systems, targets, taxonomy, technologies, telecom, test, textiles, thalassrep, thermo, tomography, tourismhosp, toxics, toxins, transplantology, transportation, traumacare, traumas, tropicalmed, universe, urbansci, uro, vaccines, vehicles, venereology, vetsci, vibration, virtualworlds, viruses, vision, waste, water, wem, wevj, wind, women, world, youth, zoonoticdis 
% For posting an early version of this manuscript as a preprint, you may use "preprints" as the journal. Changing "submit" to "accept" before posting will remove line numbers.

%---------
% article
%---------
% The default type of manuscript is "article", but can be replaced by: 
% abstract, addendum, article, book, bookreview, briefreport, casereport, comment, commentary, communication, conferenceproceedings, correction, conferencereport, entry, expressionofconcern, extendedabstract, datadescriptor, editorial, essay, erratum, hypothesis, interestingimage, obituary, opinion, projectreport, reply, retraction, review, perspective, protocol, shortnote, studyprotocol, systematicreview, supfile, technicalnote, viewpoint, guidelines, registeredreport, tutorial
% supfile = supplementary materials

%----------
% submit
%----------
% The class option "submit" will be changed to "accept" by the Editorial Office when the paper is accepted. This will only make changes to the frontpage (e.g., the logo of the journal will get visible), the headings, and the copyright information. Also, line numbering will be removed. Journal info and pagination for accepted papers will also be assigned by the Editorial Office.

%------------------
% moreauthors
%------------------
% If there is only one author the class option oneauthor should be used. Otherwise use the class option moreauthors.

%---------
% pdftex
%---------
% The option pdftex is for use with pdfLaTeX. Remove "pdftex" for (1) compiling with LaTeX & dvi2pdf (if eps figures are used) or for (2) compiling with XeLaTeX.

%=================================================================
% MDPI internal commands - do not modify
\firstpage{1}
\makeatletter
\setcounter{page}{\@firstpage}
\makeatother
\pubvolume{1}
\issuenum{1}
\articlenumber{0}
\pubyear{2024}
\copyrightyear{2024}
%\externaleditor{Academic Editor: Firstname Lastname}
\datereceived{ }
\daterevised{ } % Comment out if no revised date
\dateaccepted{ }
\datepublished{ }
%\datecorrected{} % For corrected papers: "Corrected: XXX" date in the original paper.
%\dateretracted{} % For corrected papers: "Retracted: XXX" date in the original paper.
\hreflink{https://doi.org/} % If needed use \linebreak
%\doinum{}
%\pdfoutput=1 % Uncommented for upload to arXiv.org
%\CorrStatement{yes}  % For updates


%=================================================================
% Add packages and commands here. The following packages are loaded in our class file: fontenc, inputenc, calc, indentfirst, fancyhdr, graphicx, epstopdf, lastpage, ifthen, float, amsmath, amssymb, lineno, setspace, enumitem, mathpazo, booktabs, titlesec, etoolbox, tabto, xcolor, colortbl, soul, multirow, microtype, tikz, totcount, changepage, attrib, upgreek, array, tabularx, pbox, ragged2e, tocloft, marginnote, marginfix, enotez, amsthm, natbib, hyperref, cleveref, scrextend, url, geometry, newfloat, caption, draftwatermark, seqsplit
% cleveref: load \crefname definitions after \begin{document}
\graphicspath{{figures/}}


%=================================================================
% Please use the following mathematics environments: Theorem, Lemma, Corollary, Proposition, Characterization, Property, Problem, Example, ExamplesandDefinitions, Hypothesis, Remark, Definition, Notation, Assumption
%% For proofs, please use the proof environment (the amsthm package is loaded by the MDPI class).

%=================================================================
% Full title of the paper (Capitalized)
\Title{Generative Simplex Mapping: Nonlinear Endmember Extraction and Unmixing for Hyperspectral Imagery}

% MDPI internal command: Title for citation in the left column
\TitleCitation{Generative Simplex Mapping: Nonlinear Endmember Extraction and Unmixing for Hyperspectral Imagery}

% Author Orchid ID: enter ID or remove command
\newcommand{\orcidauthorA}{0000-0002-5910-0183} % John
\newcommand{\orcidauthorB}{0000-0003-4265-9543} % David

% Authors, for the paper (add full first names)
\Author{John Waczak \orcidA{} and David J. Lary *\orcidB{}}

%\longauthorlist{yes}

% MDPI internal command: Authors, for metadata in PDF
\AuthorNames{John Waczak and David J. Lary}

% MDPI internal command: Authors, for citation in the left column
\AuthorCitation{Waczak, J.; Lary, D.J.}
% If this is a Chicago style journal: Lastname, Firstname, Firstname Lastname, and Firstname Lastname.

% Affiliations / Addresses (Add [1] after \address if there is only one affiliation.)
% Affiliations / Addresses (Add [1] after \address if there is only one affiliation.)
\address[1]{%
Hanson Center for Space Sciences, University of Texas at Dallas, Richardson, TX 75080, USA;  john.waczak@utdallas.edu (J.W.)}

% Contact information of the corresponding author
\corres{\hangafter=1 \hangindent=1.05em \hspace{-0.82em}Correspondence: david.lary@utdallas.edu} %; Tel.: (optional; include country code; if there are multiple corresponding authors, add author initials) +xx-xxxx-xxx-xxxx (F.L.)}

%\simplesumm{} % Simple summary

%\conference{} % An extended version of a conference paper

% Abstract (Do not insert blank lines, i.e. \\) 
\abstract{\textcolor{red}{UPDATE ME!}}

% Keywords
\keyword{Endmember Extraction; Spectral Unmixing; Hyperspectral Imaging; Unsupervised Machine Learning} 


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{document}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Introduction}


Hyperspectral imaging has emerged as a keystone technology in remote sensing where the ability to discern variations between high resolution spectra facilitates a plethora of critical applications from environmental monitoring and risk assessment to biodiversity convservation and sustainable agriculture. In recent years, many remote sensing platforms have been deployed with hyperspectral imaging payloads such as the Italian PRISMA mission launched in 2019 and the more recent German EnMAP launched in 2022 (\textcolor{red}{add citations}). Many future missions also plan to incorporate hyperal imaging capabilities such as the European Space Agency's CHIME which will include over $200$ bands spanning visible, near infrared (NIR), and short-wave infrared (SWIR) wavelengths (\textcolor{red}{add citations}). Additionally, the continued development of hyperspectral imaging technologies have resulted in a considerable reduction in size enabling their inclusion in the payloads of small unmanned aerial vehicles (UAV) (\textcolor{red}{add citations}). Despite these promising advances, the substantial increase in data volume generated by hyperspectral imagers as compared to their multi-spectral counterparts poses significant challenges to real-time analysis at scale.

Many approaches have been developed to make sense of hyperspectral images (HSI). For example, spectral indices such as the popular normalized difference vegetation index (NDVI) can be computed by taking ratios of spectral bands tailored to track specific reflectance characteristics (\textcolor{red}{add citations}). These indices have the advantage of being easy to compute but suffer from significant inter-instrument variability while ingoring most of the captured wavelength bins (\textcolor{red}{add citations}). An alternative approach is to pair HSI data with in-situ measurements to enable supervised models which map spectra directly to parameters of interest. However, this approach relies on serendipitous satellite overpasses above sensing sites to generate sufficient quantities of aligned data for model training and evaluation. For example, Aurin et al. combined data from over 30 years of oceanographic field campaigns with paired satellite imagery to develop robust models for the inversion of key water quality indicators such as colored dissolved organic matter (CDOM) \cite{aurin2018remote}. This approach can be accelerated by combining UAV-based hyperspectral imaging with rapid in-situ data collection using autonomous boats \cite{robot-team-1, robot-team-2}. Nevertheless, these supervised methods rely on \textit{a priori} knowledge of expected sources in order to select appropriate reference sensors for data collection. In light of these limitations, unsupervised methods are needed which can expediently extract source signatures from high dimensional HSI.

The spatial resolution of hyperspectral imagers generally results in mixed HSI pixels comprised of many individual sources called endmembers. The task of unsupervised source identification using HSI data therefore comprises two steps: endmember extraction, and abundance estimation. Techniques such as vertex component analysis (VCA), the pixel purity index (PPI), and N-FINDR, solve this first task by identifying endmember spectra from HSI assuming the presense of some pure (unmixed) pixels (\textcolor{red}{add citations}). By further assuming a linear mixing model (LMM) whereby observed spectra are described by a linear combination of endmembers with non-negative abundances, HSI can be unmixed to obtain abundance maps using a variety of techniques such as constrained least-squares. 

Once extracted, HSI can be unmixed to estimate individual endmember abundances using a variety of techniques such as constrained least-squares optimization (\textcolor{red}{add citations}). Alternatively,



- Reality: pixels captured by remote sensing platforms have spatial extents which typically cover multiple unique sources. 
- overview of spectral unmixing
- relevance to source identification in remote sensing, contaminant identication 



2. Traditional approaches for endmember extraction and spectral unmixing
- Standard assumption: linear mixing model (LMM)
- Standard requirements for LMM and other mixing models
    - abundance sum-to-one contraint $\sum_i a_i = 1$
    - endmember non-negativity constraint
- Secondary assumption: existence of pure pixels, e.g. VCA and other algorithms
- PMF and NMF as unsupervised LMM models 
- Relevance of nonlinear mixing, e.g. multiple scattering, interference effects, etc...
- Approaches for nonlinear unmixing
    - radiative transfer models
    - Deep learning approaches

3. Generative topographic mapping
- Self organizing map is a data-driven unsupervised classification algorithm
- GTM is a principled extension of the SOM to a fully probabalistic latent-space model
- Uses of the GTM in cheminformatics
- Our previous work using the GTM
- Limitations of the GTM
    - 2 dimensional latent space
    - Abundance estimation requires other heuristics like NS3
    - No direct guarantee of endmember non-negativity. In practice, this is not usually a problem when data are non-negative but it would be nice to have some guarantee
- Flexibility of the GTM model suggest augmenting the latent space distribution may lead to a more interpretable model

4. Proposed method: generative simplex mapping
- We propose a novel model dubbed Generative Simplex Mapping which naturally extends the GTM to directly facilitate mixture models. 
- The latent space is modified to an N-simplex where each vertex naturally corresponds to a unique source
- positions within the simplex correspond to different mixing ratios
- The latent space is then mapped to the data space using an activation function which allows for flexible models including both linear and nonlinear unmixing. 
- Like the NMF, the GSM simultaneously extracts endmembers \textit{and} estimates abundances
- Importantly, the GSM learns a probability distribution for the data allowing spectral 

5. Summary of paper structure
- In this paper, we present the GSM and demonstrate its application to both synthetic and real datasets. The structure of the paper is as follows: 
- In section 2 we provide a detailed overview of the GSM model and it's implementation
- In section 3 we describe experiments using synthetic and real data to evaluate the performance of the GSM
- In section 4 we present the experimental results
- In sections 5 we discuss additional applications of the GSM and further extensions to be explored in future work. 
- Finally, in section 6, we present a summary of our conclusions.

\newpage

\section{Generative Simplex Mapping}
\subsection{Method Derivation}

\begin{equation}\label{eqn:data-space-distribution}
    p(x \mid z, \mathbf{W}, \beta) = \left(\frac{\beta}{2\pi} \right)^{D/2}\exp\left( -\frac{\beta}{2}\lVert \psi(z; \mathbf{W}) - x \rVert^2 \right)
\end{equation}

\begin{equation}\label{eqn:latent-distribution}
    p(z) = \sum\limits_k^K \pi_k \, \delta(z - z_k) \quad\text{where}\quad  \sum_k^K\pi_k = 1
\end{equation}

\begin{equation}\label{eqn:llh}
    \mathcal{L}(\mathbf{W}, \beta) = \sum\limits_n^N \ln \left( \sum\limits_k^K \pi_k \, p(x_n \mid z_k, \mathbf{W}, \beta) \right)
\end{equation}

The activation function $\mathbf{\Phi}_{km} = \phi_m(z_k)$ is chosen to enable modeling linear and nonlinear mixing. For $m \leq N_v$, we take $\Phi_{km} = z_k$. For $m > N_v$, we set
\begin{equation}\label{eq:act-function}
    \mathbf{\Phi}_{km} = \begin{cases}
        \frac{s - \lVert z_k - \mu_m \rVert}{s}; & \lVert z_k - \mu_m \rVert \leq s \\
        0; & \lVert z_k - \mu_m \rVert > s
    \end{cases}
\end{equation}
The activation function centers $\mu_m$ are selected to be greater than 

We include a prior distribution on model weights to enable
\begin{align}\label{eq:weight-prior}
\begin{split}
    p(\mathbf{W} \mid \lambda_e, \lambda_w) &= \left(\frac{\lambda_e}{2\pi}\right)^{\frac{D\,N_v}{2}}\exp\left(-\frac{\lambda_e}{2}\sum\limits_d^D\sum\limits_{m=1}^{N_v}\mathbf{W}_{dm}^2 \right)\\ 
    & \qquad \cdot \left(\frac{\lambda_w}{2\pi}\right)^{\frac{D(M-N_v)}{2}}\exp\left(-\frac{\lambda_w}{2}\sum\limits_d^D\sum_{m=N_v+1}^M \mathbf{W}_{dm}^2\right)
\end{split}
\end{align}

\begin{equation}\label{eq:responsibility}
    \mathbf{R}_{kn}  = p(z_k \mid x_n, \mathbf{W}, \beta) = \dfrac{p(x_n \mid z_k, \mathbf{W}, \beta)}{\sum\limits_{k'}^K p(x_n \mid z_{k'}, \mathbf{W}, \beta)}
\end{equation}

Penalized complete-data log likelihood
\begin{adjustwidth}{-\extralength}{0cm}
\begin{equation}\label{eq:complete-data-llh}
    Q = \sum\limits_n^N\sum\limits_k^K \mathbf{R}_{kn} \left(\ln\pi_k + \frac{D}{2}\ln\left(\frac{\beta}{2\pi}\right) - \frac{\beta}{2}\sum\limits_d^D\left(\sum\limits_m^M \mathbf{W}_{dm}\mathbf{\Phi}_{km} - \mathbf{X}_{nd}\right)^2\right) + \frac{MD}{2}\ln\left(\frac{\lambda}{2\pi}\right) - \frac{\lambda}{2}\sum\limits_d^D \sum\limits_m^M \mathbf{W}_{dm}^2
\end{equation}
\end{adjustwidth}


\begin{equation}\label{eq:pi-update}
    \pi_k^{\text{new}}  = \frac{1}{N}\sum_n \mathbf{R}_{kn}
\end{equation}

\begin{equation}\label{eq:beta-update}
    \frac{1}{\beta^{\text{new}}}  = \frac{1}{ND}\sum\limits_n^N\sum\limits_k^K \mathbf{R}_{kn}\lVert \psi(z_k; \mathbf{W}) - x_n \rVert^2
\end{equation}

\begin{equation}\label{eq:W-update}
    W_{dm}^\text{new} = \dfrac{\left(X^TR^T\Phi\right)_{dm}}{\left(W\Phi^TG\Phi + \Lambda\right)_{dm}} 
\end{equation}

- multiplicative updated to guarantee non-negativity (Generalized EM procedure)
- sum-to-one constraint is guaranteed via representation of latent space with barycentric coordinates.


\begin{figure}[H]
\includegraphics[width=\columnwidth]{methods/gsm/gsm-diagram.pdf}
\caption{Illustration of the GSM. Points in the latent space (green dots) are uniformly distributed at the nodes of a regular grid on the n-simplex whose barycentric coordinates are to be interpreted as the relative abundance of n-many unique sources $z_1, ... z_{N_v}$. Here, $N_v=3$ has been chosen for illustrative purposes. These nodes are mapped into (spectral) data space via the mapping $\psi(z)$ utilizing $M$-many radially symmetric basis functions shown in red. The trained GSM also enables modelling of spectral variability via the precision parameter $\beta$ shown here as a light blue band around the estimated spectra in the data space.\label{fig:gsm-diagram}}
\end{figure}  


\subsection{Implementation Details}

The mixing coefficients are initially chosen so that $\pi_k = \frac{1}{K}$.

- Linear Method

- Nonlinear Method

- Mixed Method


\section{Experiments}
\subsection{Linear Mixing: Comparison to NMF}
- describe USGS data
- describe NMF method (for comparisson)
- describe evaluation criteria


\begin{figure}[H]
\includegraphics[width=\columnwidth]{methods/usgs/usgs-dataset.pdf}
\caption{Synthetic dataset from USGS spectra. \label{fig:usgs-data}}
\end{figure}  




\subsection{Nonlinear Mixing: Water Contaminant Identification}
- describe robot team data collection and processing
- describe 



\begin{figure}[H]
\includegraphics[width=\columnwidth]{methods/robot-team/robot-team-overview.pdf}
\caption{Robot team dataset. \label{fig:robotteam-data}}
\end{figure}  



\section{Results}
\subsection{Linear Mixing}

\begin{figure}[H]
\includegraphics[width=\columnwidth]{results/usgs/fit-comparison.pdf}
\caption{Comparison of GSM against NMF on simulated USGS dataset. \label{fig:usgs-fits}}
\end{figure}  


\begin{figure}[H]
\includegraphics[width=\columnwidth]{results/usgs/extracted-endmembers.png}
\caption{Endmembers extracted using GSM for simulated USGS dataset with SNR$=20$. \label{fig:usgs-endmembers}}
\end{figure}  


\subsection{Nonlinear Mixing: Rhodamine Dye Plume}

Exploratory data analysis: 
\begin{figure}[H]
\includegraphics[width=0.60\columnwidth]{results/robot-team/pca-variance.png}
\caption{Explained variance of PCA components for robot team dataset. A red horizontal line is superimposed on the plot indicating an explained variance of $1\%$. \label{fig:robot-team-pca}}
\end{figure}  

\begin{table}[H] 
\caption{Hyperparameter optimization: Multiple models were trained to identify optimal hyperparameter values for the GSM applied to the water spectra dataset. Here we report the top $10$ models ranked according the the BIC.}
\label{table:fit-comparison}
\begin{tabularx}{\textwidth}{CCCCCC}
\toprule
\textbf{$N_v$}	& \textbf{$\lambda_e$}	& \textbf{$\lambda_w$} & \textbf{BIC} & \textbf{AIC} & \textbf{Reconstruction RMSE}\\
\midrule
$3$ & $0.01$    & $1.0$     & $-6.195\times10^7$   & $-6.269\times10^7$   & $0.000989$ \\
$3$	& $0.001$   & $1.0$     & $-6.194\times10^7$   & $-6.268\times10^7$   & $0.000989$ \\
$3$	& $0.1$     & $1.0$	    & $-6.192\times10^7$   & $-6.265\times10^7$   & $0.000991$ \\
$3$	& $1.0$     & $1.0$	    & $-6.186\times10^7$   & $-6.260\times10^7$   & $0.001190$ \\
$4$	& $1.0$     & $1.0$	    & $-6.181\times10^7$   & $-6.255\times10^7$   & $0.001002$ \\
$4$	& $0.1$     & $1.0$	    & $-6.175\times10^7$   & $-6.249\times10^7$   & $0.001008$ \\
$4$	& $0.01$    & $1.0$     & $-6.173\times10^7$   & $-6.247\times10^7$   & $0.001009$ \\ 
$4$	& $0.001$   & $1.0$     & $-6.173\times10^7$   & $-6.247\times10^7$   & $0.001009$ \\
$4$	& $0.1$     & $10.0$    & $-6.171\times10^7$   & $-6.245\times10^7$   & $0.001011$ \\
$3$	& $1.0$     & $10.0$    & $-6.166\times10^7$   & $-6.239\times10^7$   & $0.001014$ \\
\bottomrule
\end{tabularx}
\end{table}


\begin{figure}[H]
\begin{adjustwidth}{-\extralength}{0cm}
\centering
\includegraphics[width=1.25\columnwidth]{results/robot-team/extracted-endmembers.pdf}
\end{adjustwidth}
\caption{Nonlinear GSM applied to water spectra: \textbf{(a)} Spectra corresponding to maximum abundances for each vertex in the trained GSM identified for as water, near-shore vegetation, and rhodamine dye sources. \textbf{(b)} The original hyperspectral datacube segmented according to the relative abundance of each endmember. Each water pixel defined by an NDWI $\geq 0.25$ is colored by smoothing interpolating between red, green, and blue colors corresponding to the relative abundance estimated for rhodamine, vegetation, and water spectra.}
\label{fig:robot-team-endmembers}
\end{figure}  

\newpage
\begin{figure}[H]
\begin{adjustwidth}{-\extralength}{0cm}
\centering
\includegraphics[width=\columnwidth]{results/robot-team/endmember-abundances.pdf}
\caption{Endmember distributions: \textbf{(a)} The distribution of abundance for the water class. This source dominates in the center of the water and decreases towards the edge of the pond where surface vegetation begins to dominate the reflectance signal. We note that the water abundance decreases near  the edge of the rhodamine plume reflecting dye mixing and diffusion. \textbf{(b)} The distribution of vegetation. This signal includes filamentous blue-green algae observed to accumulate in shallow waters near the shore. \textbf{(c)} The rhodamine dye plume extent segmented from the HSI. The total area for near-shore vegetation and rhodamine are estimated to be $378.6$ $\text{m}^2$ and $255.7$ $\text{m}^2$ respectively.}
\end{adjustwidth}
\label{fig:endmember-abundance-dist}
\end{figure}  
\newpage


\begin{figure}[H]
\begin{adjustwidth}{-\extralength}{0cm}
\centering
\includegraphics[width=1.25\columnwidth]{results/robot-team/plume-evo.pdf}
\end{adjustwidth}
\caption{Rhodamine plume evolution: Using the trained GSM we can track the dispersion of the rhodamine dye plume between successive drone flights. \textbf{(a)} The initial plume distribution after release. Here the dye subsumes an area of $255.7$ $\text{m}^2$. \textbf{(b)} The same plume imaged 15 minutes later now extends across an area of $571.8$ $\text{m}^2$}
\label{fig:plume-evo}
\end{figure}  


\section{Discussion}

- Comparison to NMF on linear mixing problem
    - note that there are many versions of NMF with different constraints/regularization. The purpose here was to show that GSM is at least \textit{comparable} to NMF for linear unmixing problems
    - Further note that both NMF and GSM do not rely on assumption of pure pixels (unlike VCA and other methods) 
    - GSM also allows estimation of endmember variability via the precision parameter $\beta$ i.e. nodes in data space follow Gaussian distribution with $\sigma = \sqrt{\beta^{-1}}$

- Nonlinear GSM for robot team 
    - discuss identification of rhodamine plume and easy abundance mapping via the embedding coordinate
    - discuss intrinsic dimensionality identification via abundance sparsity and BIC
    - Discuss identification of non-linearity \textit{strength} via regularization parameter and BIC.
    - reconstruction rmse can also be used to evaluate the relative performance for fixed model size (i.e. k and m) with varying regularization strength

- Limitations of the GSM
    - slow training due to computation of dinstance matrix between full dataset and embedding space
    - An incremental version of the EM procedure as described in \cite{gtm-developments} can speed this up by considering batches $X_b$ of $X$ and performing only a partial E-step by updating responsibilities corresponding to the data batch
    - However, M step still involves the "full" responsibility matrix. 
    - This can be addressed by taking an ensembling approach as described in \cite{parallel-gtm} where multiple GTM are trained on subsets of $X$ and the results are averaged. 
    - More generally, we may consider a mixture-of-GTMs as briefly desribed in \cite{gtm-orig} whereby the overall density is given as 
    \begin{equation}
        p(\mathbf{x}) = \sum_r P(r)p(\mathbf{x}\vert r)
    \end{equation}
    where there are $r$ individual GTM models with mixing coefficients $P(r)$ such that $\sum_r P(r) =1$

- Other potential extensions to GSM model
     - Adapt latent space  to allow for points whose barycentric coordinates sum to less than 1 in order to account for lighting effects (shouldn't matter for reflectance but would matter for radiance depending on incident light). This is the $\gamma$ factor that shows up in some of the mixing papers.

- Discuss other applications of the GSM
    - Optimal route planning for in-situ data collection based on prize-collecting travelling salesman problem, i.e. construct a route which maximizies area traversed in GSM latent space (e.g. the n-simplex). 
    - Applications to source apportionment, e.g. linear GSM can be used for standard source apportionment problems (no reactions) and nonlinear GSM may be useful for modeling scenarios with additional complications due to reactions and during transport from source to receptor.

\section{Conclusions}


\section{Patents}

\textcolor{red}{UPDATE ME!}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\vspace{6pt} 


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\authorcontributions{Methodology, J.W.; conceptualization, J.W.; software, J.W.; validation, J.W.; formal analysis, J.W.; investigation J.W.; resources, D.J.L.; writing---original draft preparation, J.W.; writing---review and editing, J.W. and D.J.L.; visualization, J.W.; supervision, D.J.L.; project administration, D.J.L.; funding acquisition, D.J.L. All authors have read and agreed to the published version of the manuscript.
}


\funding{\textcolor{red}{Please add: ``This research received no external funding'' or ``This research was funded by NAME OF FUNDER grant number XXX.'' and  and ``The APC was funded by XXX''. Check carefully that the details given are accurate and use the standard spelling of funding agency names at \url{https://search.crossref.org/funding}, any errors may affect your future funding.}}


\institutionalreview{Not applicable.}

\informedconsent{Not applicable.}


\dataavailability{\textcolor{red}{We encourage all authors of articles published in MDPI journals to share their research data. In this section, please provide details regarding where data supporting reported results can be found, including links to publicly archived datasets analyzed or generated during the study. Where no new data were created, or where data is unavailable due to privacy or ethical restrictions, a statement is still required. Suggested Data Availability Statements are available in section ``MDPI Research Data Policies'' at \url{https://www.mdpi.com/ethics}.}}


\acknowledgments{\textcolor{red}{In this section you can acknowledge any support given which is not covered by the author contribution or funding sections. This may include administrative and technical support, or donations in kind (e.g., materials used for experiments).}}

\conflictsofinterest{The authors declare no conflicts of interest.}
 

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Optional

%% Only for journal Encyclopedia
%\entrylink{The Link to this entry published on the encyclopedia platform.}

\abbreviations{Abbreviations}{
The following abbreviations are used in this manuscript:\\

\noindent
\begin{tabular}{@{}ll}
PRISMA & Hyperspectral Precuror of the Application Mission \\
EnMAP & Environmental Mapping and Analysis Program \\
CHIME & Copernicus Hyperspectral Imaging Mission for the Environment \\
NIR & Near Infrared \\
SWIR & Short-wave Infrared \\
UAV & Unmanned Aerial Vehicle \\
HSI & Hyperspectral Image \\
NDVI & Normalized Difference Vegetation Index \\
VCA & Vertex Component Analysis\\
PPI & Pixel Purity Index
\end{tabular}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Optional
\appendixtitles{no} % Leave argument "no" if all appendix headings stay EMPTY (then no dot is printed after "Appendix A"). If the appendix sections contain a heading then change the argument to "yes".


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{adjustwidth}{-\extralength}{0cm}
%\printendnotes[custom] % Un-comment to print a list of endnotes

\reftitle{References}

\bibliography{./references.bib}
\PublishersNote{}

\end{adjustwidth}
\end{document}

