%  LaTeX support: latex@mdpi.com 
%  For support, please attach all files needed for compiling as well as the log file, and specify your operating system, LaTeX version, and LaTeX editor.

%=================================================================
\documentclass[remotesensing,article,submit,pdftex,moreauthors]{Definitions/mdpi}

%--------------------
% Class Options:
%--------------------
%----------
% journal
%----------
% Choose between the following MDPI journals:
% acoustics, actuators, addictions, admsci, adolescents, aerobiology, aerospace, agriculture, agriengineering, agrochemicals, agronomy, ai, air, algorithms, allergies, alloys, analytica, analytics, anatomia, animals, antibiotics, antibodies, antioxidants, applbiosci, appliedchem, appliedmath, applmech, applmicrobiol, applnano, applsci, aquacj, architecture, arm, arthropoda, arts, asc, asi, astronomy, atmosphere, atoms, audiolres, automation, axioms, bacteria, batteries, bdcc, behavsci, beverages, biochem, bioengineering, biologics, biology, biomass, biomechanics, biomed, biomedicines, biomedinformatics, biomimetics, biomolecules, biophysica, biosensors, biotech, birds, bloods, blsf, brainsci, breath, buildings, businesses, cancers, carbon, cardiogenetics, catalysts, cells, ceramics, challenges, chemengineering, chemistry, chemosensors, chemproc, children, chips, cimb, civileng, cleantechnol, climate, clinpract, clockssleep, cmd, coasts, coatings, colloids, colorants, commodities, compounds, computation, computers, condensedmatter, conservation, constrmater, cosmetics, covid, crops, cryptography, crystals, csmf, ctn, curroncol, cyber, dairy, data, ddc, dentistry, dermato, dermatopathology, designs, devices, diabetology, diagnostics, dietetics, digital, disabilities, diseases, diversity, dna, drones, dynamics, earth, ebj, ecologies, econometrics, economies, education, ejihpe, electricity, electrochem, electronicmat, electronics, encyclopedia, endocrines, energies, eng, engproc, entomology, entropy, environments, environsciproc, epidemiologia, epigenomes, est, fermentation, fibers, fintech, fire, fishes, fluids, foods, forecasting, forensicsci, forests, foundations, fractalfract, fuels, future, futureinternet, futurepharmacol, futurephys, futuretransp, galaxies, games, gases, gastroent, gastrointestdisord, gels, genealogy, genes, geographies, geohazards, geomatics, geosciences, geotechnics, geriatrics, grasses, gucdd, hazardousmatters, healthcare, hearts, hemato, hematolrep, heritage, higheredu, highthroughput, histories, horticulturae, hospitals, humanities, humans, hydrobiology, hydrogen, hydrology, hygiene, idr, ijerph, ijfs, ijgi, ijms, ijns, ijpb, ijtm, ijtpp, ime, immuno, informatics, information, infrastructures, inorganics, insects, instruments, inventions, iot, j, jal, jcdd, jcm, jcp, jcs, jcto, jdb, jeta, jfb, jfmk, jimaging, jintelligence, jlpea, jmmp, jmp, jmse, jne, jnt, jof, joitmc, jor, journalmedia, jox, jpm, jrfm, jsan, jtaer, jvd, jzbg, kidneydial, kinasesphosphatases, knowledge, land, languages, laws, life, liquids, literature, livers, logics, logistics, lubricants, lymphatics, machines, macromol, magnetism, magnetochemistry, make, marinedrugs, materials, materproc, mathematics, mca, measurements, medicina, medicines, medsci, membranes, merits, metabolites, metals, meteorology, methane, metrology, micro, microarrays, microbiolres, micromachines, microorganisms, microplastics, minerals, mining, modelling, molbank, molecules, mps, msf, mti, muscles, nanoenergyadv, nanomanufacturing,\gdef\@continuouspages{yes}} nanomaterials, ncrna, ndt, network, neuroglia, neurolint, neurosci, nitrogen, notspecified, %%nri, nursrep, nutraceuticals, nutrients, obesities, oceans, ohbm, onco, %oncopathology, optics, oral, organics, organoids, osteology, oxygen, parasites, parasitologia, particles, pathogens, pathophysiology, pediatrrep, pharmaceuticals, pharmaceutics, pharmacoepidemiology,\gdef\@ISSN{2813-0618}\gdef\@continuous pharmacy, philosophies, photochem, photonics, phycology, physchem, physics, physiologia, plants, plasma, platforms, pollutants, polymers, polysaccharides, poultry, powders, preprints, proceedings, processes, prosthesis, proteomes, psf, psych, psychiatryint, psychoactives, publications, quantumrep, quaternary, qubs, radiation, reactions, receptors, recycling, regeneration, religions, remotesensing, reports, reprodmed, resources, rheumato, risks, robotics, ruminants, safety, sci, scipharm, sclerosis, seeds, sensors, separations, sexes, signals, sinusitis, skins, smartcities, sna, societies, socsci, software, soilsystems, solar, solids, spectroscj, sports, standards, stats, std, stresses, surfaces, surgeries, suschem, sustainability, symmetry, synbio, systems, targets, taxonomy, technologies, telecom, test, textiles, thalassrep, thermo, tomography, tourismhosp, toxics, toxins, transplantology, transportation, traumacare, traumas, tropicalmed, universe, urbansci, uro, vaccines, vehicles, venereology, vetsci, vibration, virtualworlds, viruses, vision, waste, water, wem, wevj, wind, women, world, youth, zoonoticdis 
% For posting an early version of this manuscript as a preprint, you may use "preprints" as the journal. Changing "submit" to "accept" before posting will remove line numbers.

%---------
% article
%---------
% The default type of manuscript is "article", but can be replaced by: 
% abstract, addendum, article, book, bookreview, briefreport, casereport, comment, commentary, communication, conferenceproceedings, correction, conferencereport, entry, expressionofconcern, extendedabstract, datadescriptor, editorial, essay, erratum, hypothesis, interestingimage, obituary, opinion, projectreport, reply, retraction, review, perspective, protocol, shortnote, studyprotocol, systematicreview, supfile, technicalnote, viewpoint, guidelines, registeredreport, tutorial
% supfile = supplementary materials

%----------
% submit
%----------
% The class option "submit" will be changed to "accept" by the Editorial Office when the paper is accepted. This will only make changes to the frontpage (e.g., the logo of the journal will get visible), the headings, and the copyright information. Also, line numbering will be removed. Journal info and pagination for accepted papers will also be assigned by the Editorial Office.

%------------------
% moreauthors
%------------------
% If there is only one author the class option oneauthor should be used. Otherwise use the class option moreauthors.

%---------
% pdftex
%---------
% The option pdftex is for use with pdfLaTeX. Remove "pdftex" for (1) compiling with LaTeX & dvi2pdf (if eps figures are used) or for (2) compiling with XeLaTeX.

%=================================================================
% MDPI internal commands - do not modify
\firstpage{1}
\makeatletter
\setcounter{page}{\@firstpage}
\makeatother
\pubvolume{1}
\issuenum{1}
\articlenumber{0}
\pubyear{2024}
\copyrightyear{2024}
%\externaleditor{Academic Editor: Firstname Lastname}
\datereceived{ }
\daterevised{ } % Comment out if no revised date
\dateaccepted{ }
\datepublished{ }
%\datecorrected{} % For corrected papers: "Corrected: XXX" date in the original paper.
%\dateretracted{} % For corrected papers: "Retracted: XXX" date in the original paper.
\hreflink{https://doi.org/} % If needed use \linebreak
%\doinum{}
%\pdfoutput=1 % Uncommented for upload to arXiv.org
%\CorrStatement{yes}  % For updates


%=================================================================
% Add packages and commands here. The following packages are loaded in our class file: fontenc, inputenc, calc, indentfirst, fancyhdr, graphicx, epstopdf, lastpage, ifthen, float, amsmath, amssymb, lineno, setspace, enumitem, mathpazo, booktabs, titlesec, etoolbox, tabto, xcolor, colortbl, soul, multirow, microtype, tikz, totcount, changepage, attrib, upgreek, array, tabularx, pbox, ragged2e, tocloft, marginnote, marginfix, enotez, amsthm, natbib, hyperref, cleveref, scrextend, url, geometry, newfloat, caption, draftwatermark, seqsplit
% cleveref: load \crefname definitions after \begin{document}
\graphicspath{{figures/}}


%=================================================================
% Please use the following mathematics environments: Theorem, Lemma, Corollary, Proposition, Characterization, Property, Problem, Example, ExamplesandDefinitions, Hypothesis, Remark, Definition, Notation, Assumption
%% For proofs, please use the proof environment (the amsthm package is loaded by the MDPI class).

%=================================================================
% Full title of the paper (Capitalized)
\Title{Generative Simplex Mapping: Nonlinear Endmember Extraction for Source Apportionment and Spectral Unmixing for Hyperspectral Imagery}
% \Title{Generative Simplex Mapping: Nonlinear Endmember Extraction and Spectral Unmixing for Hyperspectral Imagery}

% MDPI internal command: Title for citation in the left column
\TitleCitation{Generative Simplex Mapping: Nonlinear Endmember Extraction and Spectral Unmixing for Hyperspectral Imagery}

% Author Orchid ID: enter ID or remove command
\newcommand{\orcidauthorA}{0000-0002-5910-0183} % John
\newcommand{\orcidauthorB}{0000-0003-4265-9543} % David

% Authors, for the paper (add full first names)
\Author{John Waczak \orcidA{} and David J. Lary *\orcidB{}}

%\longauthorlist{yes}

% MDPI internal command: Authors, for metadata in PDF
\AuthorNames{John Waczak and David J. Lary}

% MDPI internal command: Authors, for citation in the left column
\AuthorCitation{Waczak, J.; Lary, D.J.}
% If this is a Chicago style journal: Lastname, Firstname, Firstname Lastname, and Firstname Lastname.

% Affiliations / Addresses (Add [1] after \address if there is only one affiliation.)
% Affiliations / Addresses (Add [1] after \address if there is only one affiliation.)
\address[1]{%
Hanson Center for Space Sciences, University of Texas at Dallas, Richardson, TX 75080, USA;  john.waczak@utdallas.edu (J.W.)}

% Contact information of the corresponding author
\corres{\hangafter=1 \hangindent=1.05em \hspace{-0.82em}Correspondence: david.lary@utdallas.edu} %; Tel.: (optional; include country code; if there are multiple corresponding authors, add author initials) +xx-xxxx-xxx-xxxx (F.L.)}

%\simplesumm{} % Simple summary

%\conference{} % An extended version of a conference paper

% Abstract (Do not insert blank lines, i.e. \\) 
\abstract{\textcolor{red}{UPDATE ME!}}

% Keywords
\keyword{Endmember Extraction; Spectral Unmixing; Hyperspectral Imaging; Unsupervised Machine Learning; Source Apportionment} 


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{document}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Introduction}


Hyperspectral imaging has emerged as a keystone technology in remote sensing where the ability to discern variations between high resolution spectra supports a plethora of critical applications such as environmental monitoring, biodiversity conservation, sustainable agriculture, and more. In recent years, many remote sensing platforms have been deployed with hyperspectral imaging payloads such as the Italian PRISMA mission launched in 2019, the German EnMAP launched in 2022, and recently NASA's PACE satellite launched in 2024 \cite{PRISMA-orig, EnMAP-orig, PACE-orig}. Many future missions also plan to incorporate hyperspectral imaging capabilities such as the European Space Agency's CHIME which will include over $200$ bands spanning visible, near infrared (NIR), and short-wave infrared (SWIR) wavelengths \cite{CHIME-orig}. The continued development of hyperspectral imaging technology has also led to a considerable reduction in size enabling their inclusion in the payloads of small unmanned aerial vehicles (UAV) \cite{adao2017hyperspectral, arroyo2019implementation}. Despite the proliferation of hyperspectral imaging, the considerable increase in data volume associated with hyperspectral images (HSI) poses significant challenges to real-time analysis at scale.

Many approaches have been developed to make sense of HSI data. For example, spectral indices like the popular normalized difference vegetation index (NDVI) can be computed by taking ratios of spectral bands tailored to track specific reflectance characteristics \cite{thenkabail-indices,thenkabail2018hyperspectral}. These indices have the advantage of being easy to compute but suffer from significant inter-instrument variability while ignoring most of the information captured in HSI spectra \cite{ndvi-variability}. An alternative approach is to pair HSI data with in-situ measurements to enable supervised models which map spectra directly to parameters of interest. However, this approach relies on serendipitous satellite overpasses above sensing sites to generate sufficient quantities of aligned data for model training and evaluation. For instance, Aurin et al. combined data from over 30 years of oceanographic field campaigns with paired satellite imagery to develop robust models for the inversion of key water quality indicators such as colored dissolved organic matter (CDOM) \cite{aurin2018remote}. This approach can be accelerated by combining UAV-based hyperspectral imaging with rapid in-situ data collection using autonomous boats \cite{robot-team-1, robot-team-2}. Nevertheless, these supervised methods rely on \textit{a priori} knowledge of expected sources in order to identify appropriate reference sensors for data collection. In light of these limitations, unsupervised methods are needed which can expediently extract source signatures from high dimensional HSI.

The spatial resolution of hyperspectral imagers generally results in  pixels with mixed signals from multiple sources called endmembers. The task of unsupervised source identification using HSI data therefore involves two steps: endmember extraction, and abundance estimation. Techniques such as vertex component analysis (VCA), the pixel purity index (PPI), and N-FINDR solve this first task by identifying endmember spectra from HSI assuming the presence of some pure (unmixed) pixels \cite{vca-orig, ppi-orig, N-FINDR-orig}. By further assuming a linear mixing model (LMM) whereby observed spectra are described by a linear combination of endmembers with non-negative abundances, HSI can then be unmixed using a variety of techniques such as constrained least-squares \cite{spectral-unmixing-orig, fcls-unmixing}. Among these methods Non-negative Matrix Factorization (NMF) is a widely used approach which extracts endmember spectra and unmixes abundances simultaneously via matrix factorization \cite{nmf-orig, unmixing-nmf-review, unmixing-nmf-review-2}. The update equations for NMF can be formulated into multiplicative updates which guarantee the non-negativity of endmember spectra and their associated abundances \cite{nmf-algorithms}.  For this reason the continued development of new NMF varieties remains an active area of research.

In realistic scenes multiple scattering and surface variability can easily challenge the assumption of linear mixing \cite{heylen2014review}. Water-based HSI specifically are prone to nonlinear mixing effects due to absorption features of dissolved and suspended substances, fluorescence of organic matter, and particulate scattering in turbid waters \cite{hsi-absorption, hsi-fluorescence, hsi-turibidity}. With the growing popularity of deep learning approaches in remote sensing, a variety of models based on autoencoder architectures have been introduced which enable nonlinear unmixing of HSI data \cite{non-negative-autoencoders,su2019daen,palsson2020convolutional,}. However, the complexity introduced by these models significantly impacts training time and decreases model interpretability. An ideal approach should enable both endmember extraction and nonlinear unmixing while accounting for spectral variability. 

The self-organizing map (SOM) is an unsupervised machine learning method which maps high-dimensional data onto a low-dimensional grid while preserving the topological relationships between data points \cite{kohonen-som-1}. This low dimensional representation provides a convenient way to visualize HSI data while the weight vectors for each SOM node can be interpreted as representative spectra \cite{cantero2004analysis, duran2007time,som-hsi}. If labelled reference spectra are available, the SOM can be used to enable semi-supervised labelling of HSI pixels \cite{riese2019supervised}. The SOM has also been shown to be effective for compression of HSIs acquired by a CubeSat \cite{som-satellite}. Despite these capabilities, the SOM does not offer a probabilistic interpretation and relies on a heuristic training procedure with hyperparameters that can be challenging to tune. To address these shortcomings, Bishop et al introduced the Generative Topographic Mapping (GTM), a probabilistic latent-variable model inspired by the SOM \cite{gtm-orig}. When the latent space is chosen to be two dimensional, the GTM can be used to visualize the distribution of HSI spectra while the mapping of latent space nodes into the HSI data space provides endmembers\cite{robot-team-gtm}. Unfortunately, the rectangular latent space grid employed by the GTM does not directly translate into endmember abundances. Additionally, the expectation-maximization (EM) algorithm used to train the GTM does not guarantee non-negativity of GTM node spectra. 

In this paper, we introduce a new variant of the GTM dubbed Generative Simplex Mapping (GSM) which can extract endmember spectra and unmix nonlinear mixtures. By replacing the rectangular latent space of the GTM with a gridded $n$-simplex, the vertices of the GSM can be immediately interpreted as endmembers corresponding to $n+1$ unique spectral signatures. The mapping from the latent space to the HSI data space models signal mixing while barycentric coordinates for the latent space simplex estimate relative endmember abundances. Furthermore, by taking inspiration from the multiplicative updates of NMF, the GSM algorithm maintains non-negativity of resulting endmember spectra. If only linear mixing is present, the GSM algorithm drives nonlinear contributions to $0$. Prior distributions included for GSM model weights yield hyperparameters which can be tuned to control the smoothness of resulting spectra and the degree of nonlinear mixing applied.

The rest of the paper is structured as follows. Section~\ref{sec:gsm} describes the proposed method. Section~\ref{sec:experiments} and Section~\ref{sec:results} describe experiments using simulated and real HSI data to evaluate the GSM model. Section~\ref{sec:disucssion} discusses additional applications and extensions of the GSM to be explored in future work. Finally, Section~\ref{sec:conclusions} finishes the paper with some closing remarks.



\section{Generative Simplex Mapping}\label{sec:gsm}

In the original GTM formulation, data vectors $x$ (reflectance spectra) are described by latent variables $z$ mapped into the data space by a nonlinear function $\psi$. The data space distribution is taken to be normal with precision parameter $\beta$ to account for measurement noise and spectral variability. The GSM uses same structure, that is
\begin{equation}\label{eqn:data-space-distribution}
    p(x \mid z, \mathbf{W}, \beta) = \left(\frac{\beta}{2\pi} \right)^{D/2}\exp\left( -\frac{\beta}{2}\lVert \psi(z; \mathbf{W}) - x \rVert^2 \right)
\end{equation}
where $\mathbf{W}$ are model weights which parameterize the mapping $\psi$.

Assuming data are uniformly sampled from an embedded manifold in the data space, the GTM models the latent space using a rectangular grid with $K$-many nodes of equal prior probability. To adapt the GTM to describe endmember mixing, the GSM makes two key changes. The first is to replace the rectangular GTM grid with a gridded simplex having $N_v$ vertices. Barycentric coordinates for each node then describe the relative abundance of each endmember with each vertex corresponding to a pure endmember. The second change is to replace the equal prior probabilities with adaptive mixing coefficients $\pi_k$ to allow the GSM to model HSI with non-uniform mixing distributions. Together, this leads to a latent space prior distribution given by
\begin{equation}\label{eqn:latent-distribution}
    p(z) = \sum\limits_k^K \pi_k \, \delta(z - z_k) \quad\text{where}\quad  \sum_k^K\pi_k = 1.
\end{equation}

For a dataset containing $N$-many records, these definitions yield a log likelihood function given by
\begin{equation}\label{eqn:llh}
    \mathcal{L} = \sum\limits_n^N \ln \left( \sum\limits_k^K \pi_k \, p(x_n \mid z_k, \mathbf{W}, \beta) \right)
\end{equation}
which can be maximized to obtain optimal values for model weights $\mathbf{W}$, mixing coefficients $\pi_k$, and precision $\beta$. Rather than optimizing Eq~\ref{eqn:llh} directly, we instead choose a particular form for $\psi$ to allow fitting the GSM via an EM algorithm.

In the standard LMM model, the mapping $\psi$ is given by $\psi(z;W) = \mathbf{W}z$ where the columns of $\mathbf{W}$ correspond to endmember spectra. To model nonlinear mixing, $z$ is replaced by the output of $M$-many activation functions such that $\psi(z;W) = \mathbf{W}\phi(z)$. The activations applied to each GSM node can then be collected to form a matrix with elements $\Phi_{km} = \phi_m(z_k)$. For $m \leq N_v$, we take $\Phi_{km} = z_k$ to model linear mixing. The remaining  $M-N_v$ activations are computed using radial basis functions (RBF) with centers $\mu_m$ distributed throughout the simplex (but not at the vertices) with 
\begin{equation}\label{eq:act-function}
    \Phi_{km} = \begin{cases}
        \dfrac{s - \lVert z_k - \mu_m \rVert}{s}; & \lVert z_k - \mu_m \rVert \leq s \\
        0; & \lVert z_k - \mu_m \rVert > s
    \end{cases}
\end{equation}
where $s$ is the spacing between RBF centers. In this form, the first $N_v$ columns of $\mathbf{W}$ correspond to endmember spectra while the remaining columns account for additional nonlinear effects. For linear mixing, the GSM training algorithm should therefore drive $W_{dm}$ to $0$ for $m\geq N_v$. A visualization of the GSM model is shown in Fig~\ref{fig:gsm-diagram}.

\begin{figure}[H]
\includegraphics[width=\columnwidth]{methods/gsm/gsm-diagram.pdf}
\caption{Illustration of the GSM. The latent space consists of a grid of $K$-many points (green dots) distributed throughout a simplex with $N_v$ vertices. Barycentric coordinates of each node in the simplex correspond to the relative abundance of $N_v$-many unique sources. Here, $N_v=3$ has been chosen for illustrative purposes. Nodes are mapped into the data space via the map $\psi(z)$ utilizing $M$-many radially symmetric basis functions (red). Spectral variability is estimated via the precision parameter $\beta$ shown here in the data space as a light blue band around the spectrum given by $\psi(z)$.}
\label{fig:gsm-diagram}
\end{figure}  

To further constrain the model, we introduce prior distributions on the weights $\mathbf{W}$. For $m\leq N_v$ we take $W_{dm}\sim\mathcal{N}(0, \lambda_e^{-1})$ corresponding to a zero-mean Gaussian with variance $\lambda_e^{-1}$. For $m>N_v$ we use a zero-mean Laplace distribution, $W_{dm}\sim\dfrac{\lambda_w}{2}\exp(-\lambda_w\lvert W_{dm}\rvert)$, with scale parameter $\lambda_w^{-1}$. Under these choices $\lambda_e$ corresponds to $L_2$ regularization on endmember spectra while $\lambda_w$ corresponds to $L_1$ regularization on the nonlinear activations. In other words, $\lambda_e$ governs the smoothness of the resulting endmembers while $\lambda_w$ encourages sparsity for the nonlinear contributions.

An EM algorithm for the GSM model can now be formulated as follows. Suppose that we have current estimates for the model weights $\mathbf{W}$, mixing coefficients $\pi_k$, and precision parameter $\beta$. During the expectation step we compute the posterior probabilities, that is, the responsibility of each GSM node for each spectrum in the dataset:
\begin{equation}\label{eq:responsibility}
    R_{kn}  = p(z_k \mid x_n, \mathbf{W}, \beta) = \dfrac{\pi_k \, p(x_n \mid z_k, \mathbf{W}, \beta)}{\sum\limits_{k'}^K \pi_{k'} \, p(x_n \mid z_{k'}, \mathbf{W}, \beta)}.
\end{equation}
For the maximization step, we consider the expectation of the penalized complete-data log likelihood given by
\begin{equation}\label{eq:complete-data-llh}
\begin{aligned}
    Q &= \sum_n^N\sum_k^K R_{kn} \left(\ln\pi_k + \frac{D}{2}\ln\left(\frac{\beta}{2\pi}\right) - \frac{\beta}{2}\sum_d^D\left(\sum_m^M W_{dm}\Phi_{km} - X_{nd}\right)^2\right) \\ 
    &\qquad + \frac{N_vD}{2}\ln\left(\frac{\lambda_e}{2\pi}\right) - \frac{\lambda}{2}\sum_d^D \sum_{m=1}^{N_v} W_{dm}^2  \\ 
    &\qquad + (M-N_v)D\ln\left(\dfrac{\lambda_w}{2}\right) - \lambda_w\sum_d^D\sum_{m=N_v+1}^{M} W_{dm}
\end{aligned}
\end{equation}
where $X_{nd}$ is the $d$-th component of the $n$-th spectrum in the dataset. Eq~\ref{eq:complete-data-llh} is then maximized with respect to $\pi_k$, $\beta$, and $\mathbf{W}$ to obtain new parameter values. A detailed overview of the EM procedure can be found in Bishop \cite{bishop-prml}.

For $\pi_k$, optimization can be performed using Lagrange multipliers to maintain the condition that $\sum_k\pi_k=1$. Doing so yields
\begin{equation}\label{eq:pi-update}
    \pi_k^{\text{new}}  = \frac{1}{N}\sum_n R_{kn}
\end{equation}

Optimization of Eq~\ref{eq:complete-data-llh} with respect to $\mathbf{W}$ leads to a linear system which can be solved using standard numerical methods. However, in this form we can not guarantee the non-negativity of $\mathbf{W}$ required to describe reflectance spectra. We therefore take inspiration from the multiplicative updates for NMF introduced by Lee and Seung \cite{nmf-orig}. A standard gradient-based update for $\mathbf{W}$ would take the form
\begin{equation}
    \mathbf{W}_{\text{new}} = \mathbf{W} + \eta\frac{\partial Q}{\partial \mathbf{W}}
\end{equation}
for some learning rate $\eta$. Therefore, we differentiate to obtain
\begin{equation}
    \frac{\partial Q}{\partial \mathbf{W}} = -\beta \mathbf{W}\mathbf{\Phi}^T\mathbf{G}\mathbf{\Phi} - \mathbf{\Lambda} + \beta \mathbf{X}^T\mathbf{R}^T\mathbf{\Phi}
\end{equation}
where $\mathbf{G}$ is a diagonal matrix with $G_{kk} = \sum_n R_{kn}$ and $\mathbf{\Lambda}$ is given by 
\begin{equation}
    \Lambda_{dm} = \begin{cases}
        \lambda_e W_{dm}; & m \leq N_v \\ 
        \lambda_w; & m > N_v
    \end{cases}
\end{equation}
If we allow for individual learning rates $\eta_{dm}$ for each element of $\mathbf{W}$, then choosing 
\begin{equation}
    \eta_{dm} = \frac{W_{dm}}{\left(\beta \mathbf{W}\mathbf{\Phi}^T\mathbf{G}\mathbf{\Phi}\right)_{dm} + \Lambda_{dm}}
\end{equation}
results in a multiplicative update rule given by
\begin{equation}\label{eq:W-update}
    W_{dm}^{new}  = W_{dm} \cdot \dfrac{\left(\beta \mathbf{X}^T\mathbf{R}^T\mathbf{\Phi}\right)_{dm}}{\left(\beta \mathbf{W}\mathbf{\Phi}^T\mathbf{G}\mathbf{\Phi}\right)_{dm} + \Lambda_{dm}}
\end{equation}
From Eq~\ref{eq:W-update}, it is clear that we are multiplying $W_{dm}$ by strictly non-negative values, and therefore, non-negative $W_{dm}$ will remain so during each update. This update can also be repeated multiple times during each M-step to accelerate convergence.

Lastly, optimizing Eq~\ref{eq:complete-data-llh} with respect to $\beta$ yields
\begin{equation}\label{eq:beta-update}
    \frac{1}{\beta^{\text{new}}}  = \frac{1}{ND}\sum\limits_n^N\sum\limits_k^K R_{kn}\lVert \psi(z_k; \mathbf{W}) - x_n \rVert^2.
\end{equation}

To train a GSM model, weights $\mathbf{W}$ are randomly initialized to positive values. The mixing coefficients are initially set to $\pi_k = 1/K$. Finally, the precision parameter $\beta$ is initialized to the variance of the $(N_v+1)$-th principal component. After initialization, the expectation and maximization steps are repeated in turn until $Q$ converges to a predetermined tolerance level. For large $N_v$ we note that generating a regular grid within the simplex becomes cumbersome as the number grid nodes scales as ${k + N_v - 2 \choose N_v - 1}$ for $k$ nodes per edge. An alternative approach is to randomly sample points within the simplex to obtain a total of $K$ nodes. A Dirichlet distribution 
\begin{equation}
    p(z_1,...z_{N_v}) = \frac{\Gamma(\sum_i^{N_v}\alpha_i)}{\prod_{i}^{N_v}\Gamma(\alpha_i)} \prod_{i}^{N_v} z_i^{\alpha_i-1}
\end{equation}
with all $\alpha_i=1$ can be used to uniformly sample within the simplex. Since the mixing coefficients $\pi_k$ are adaptive, variability in node separation should not significantly impact the resulting GSM.

The probabilistic form of the GSM means that a variety of information criteria can be used to evaluate the model fits. In this paper we consider two metrics, the Bayesian Information Criterion (BIC),
\begin{equation}
   \text{BIC} = P\ln(N) - 2\mathcal{L},
\end{equation}
and the Akaike Information Criterion (AIC),
\begin{equation}
    \text{AIC} = 2P - 2\mathcal{L}
\end{equation}
where $P$ is the total number of model parameters and $\mathcal{L}$ is the log likelihood from Eq~\ref{eqn:llh}.

The map $\psi$ provides the representation of each node $z_k$ in the data space. Importantly, applying $\psi$ to each vertex extracts endmember spectra from the GSM. Slices $R_{\left[:,n\right]}$ of the matrix $\mathbf{R}$ define the responsibility of each latent node $z_k$ for the $n$-th spectrum $x_n$ in the dataset. Therefore once the GSM has been trained, $\mathbf{R}$ can be used to unmix endmember abundances by representing each record in the latent space via the mean: 
\begin{equation}
    \hat{z}_n = \sum_k^K R_{kn}z_k.
\end{equation}

A freely available implementation of the GSM is accessible at \cite{gtm-code}. The code is written in the Julia programming language and follows the Machine Learning in Julia (MLJ) common interface \cite{bezanson2012julia, blaom2020mlj}.

\section{Experiments}\label{sec:experiments}

\subsection{Linear Mixing: Comparison to NMF}

To illustrate the effectiveness of the GSM, we first demonstrate its ability to model simple linear mixing by driving nonlinear weights to zero during the fitting process. To this end, a synthetic dataset comprising linear mixtures of three sample spectra from the U.S. Geological Survey digital spectral library was generated by sampling $1000$ abundance vectors from a Dirichlet distribution with $\alpha_1=\alpha_2=\alpha_3=1/3$ \cite{usgs-spectra}. These known spectra and abundances are visualized in Figure~\ref{fig:usgs-data}.

\begin{figure}[H]
\includegraphics[width=\columnwidth]{methods/usgs/usgs-dataset.pdf}
\caption{Synthetic dataset fromed from USGS spectra. \textbf{(a)} spectra from the USGS spectral database used as the ground truth endmembers. These spectra were selected to match \cite{vca-orig}. \textbf{(b)} The distribution of abundances used in the dataset. Samples were generated using a Dirichlet distribution with $\alpha_1=\alpha_2=\alpha_3=1/3$.}
\label{fig:usgs-data}
\end{figure}  

Zero-mean Gaussian noise was added to the data to yield $9$ datasets with signal-to-noise ratios (SNR) ranging from $0$ to $\infty$ to examine the impact of random noise on GSM performance. For each of these datasets, a GSM was trained using $25$ nodes per edge with $\lambda_e$ set to the default value of $0.01$ and fixed to $\lambda_w = 100$. The large value of $\lambda_w$ was chosen to encourage the GSM to prefer models with limited nonlinear mixing.

For comparison, NMF models were also trained on each dataset as this method is both highly popular and does not include the pure-pixel assumption common to other technqieus like VCA and PPI. Countless variations on the original NMF method have been introduced into the literature with one review identifying more than $100$ distinct NMF variations \cite{unmixing-nmf-review}. For the purpose of evaluating the GSM, we considered the standard $\ell_2$ and KL-divergence formulations introduced by Lee and Seung \cite{nmf-algorithms}. We also consider the robust $\ell_{2,1}$ NMF as described by Kong et al \cite{nmf-l21}.

Four metrics were used to compare model performance. For endmember extraction the mean spectral angle and mean RMSE between true endmembers $\rho_i$ and extracted endmembers $\hat{\rho}_i$ were computed where the spectral angle for the $i$-th endmember is defined as 
\begin{equation}
    \theta(\rho_i, \hat{\rho}_i) =  \arcos\left( \dfrac{\langle \rho_i, \hat{\rho}_i \rangle}{\lVert \rho_i \rVert \cdot \lVert \hat{\rho}_i \rVert}\right)
\end{equation}
and the RMSE for the $i$-th endmember is
\begin{equation}
    \text{RMSE}(\rho_i, \hat{\rho}_i) = \sqrt{\frac{1}{D-1}\sum_d^D\left(\rho_i(\lambda_d) - \hat{\rho}_i(\lambda_d) \right)^2}.
\end{equation}
Abundance estimation was similarly evaluated using the Mean RMSE between true abundance and estimated abundance values for each endmember. Finally, the reconstruction RMSE, that is, the RMSE computed between the original dataset and the dataset reconstructed via the extracted endmembers and their associated abundances was computed. This provides a model-agnostic criterion to guarantee that each model sufficiently converged during training.

\subsection{Nonlinear Mixing: Water Contaminant Identification}

To assess the ability of the GSM to unmix realistic scenes likely to involve nonlinear mixing effects, we consider a dataset of real HSI collected in Montague, North Texas on 9 December 2020. A Freefly Alta-X autonomous quadcopter was used as the UAV platform and was equipped with a Resonon Pika XC2 visible+near-infrared (VNIR) hyperspectral imager to acquire multiple HSI. Each HSI pixel included 462 wavelength bins ranging from $391$ to $1011$ nm. To evaluate the ability of the GSM to identify potential contaminant sources, rhodamine dye, a commonly used tracer in hydrological studies, was released into a pond. Two UAV flights spaced 15 minutes apart were used to capture the evolution of the plume as it dispersed into the surrounding water.

The hyperspectral imager is in a pushbroom configuration so that HSI were captured one scan-line at a time. Data from an embedded GPS/INS unit enable direct georectification of captured imagery using the method outlined in \cite{muller2002program}. Additionally, an upward-facing Ocean Optics UV-Vis NIR spectrometer with a cosine corrector was included on the top of the UAV to measure incident solar irradiance. The configuration of the UAV together with a sample HSI are shown in Figure~\ref{fig:robotteam-data}. For a more detailed description of the system, the reader is directed to ref. \cite{robot-team-1, robot-team-2}.

\begin{figure}[t]
\includegraphics[width=\columnwidth]{methods/robot-team/robot-team-overview.pdf}
\caption{Real HSI Dataset. \textbf{(a)} the UAV used to collect hyperspectral images. \textbf{(b)} The Resonon Pika XC2 hyperspectral imager used to acquire HSI. \textbf{(c)} A sample hyperspectral data cube. Spectra a plotted using at their geographic position with the log10-reflectance colored along the z axis and a pseudocolor image on top. The signature of the rhodamine dye plume is clearly identifiable in the water. \label{fig:robotteam-data}}
\end{figure}  

Raw HSI were converted to reflectance using the downwelling irradiance spectrum captured simultaneously with each HSI. Given the UAV flies with the imager oriented to nadir, the reflectance is then given by 
\begin{equation}
    \rho(\lambda) = \pi\,L(\lambda)/E_d(\lambda)
\end{equation}
where $L$ is the spectral radiance, $E_d$ is the downwelling irradiance, and a factor of $\pi$ is included resulting from the assumption of diffuse upwelling radiance \cite{ruddick2019review}. The UAV flights were performed near solar noon to maximize the amount of sunlight illuminating the water. For this Pond in North Texas, this corresponded to an average solar zenith angle of $56.7^{\circ}$ resulting in HSI with negligible sunglint effects.

From the collected HSI, a water-only pixel mask was generated by identifying all pixel spectra with a normalized difference water index (NDWI) above $0.25$ as defined in ref. \cite{ndwi}. Of these water pixels, a combined dataset of $15,000$ spectra were sampled for GSM training. As a final processing step, reflectance spectra were limited to $\lambda \leq 900$ nm as wavelengths above this threshold showed significant noise.

In order to justify values for the appropriate number of endmembers, the final dataset was decomposed using principal component analysis (PCA). A plot of the sorted explained variance for each PCA component is shown in Figure~\ref{fig:robot-team-pca}
\begin{figure}[H]
\begin{center}
\includegraphics[width=0.60\columnwidth]{results/robot-team/pca-variance.png}
\end{center}
\vspace{-20pt}
\caption{Explained variance of PCA components for the real HSI dataset. A red horizontal line is superimposed on the plot marking an explained variance of $1\%$. All components past the fourth explain less than $1\%$ of the observed variance.}
\label{fig:robot-team-pca}
\end{figure}  
The PCA decomposition of the data suggests that at least $3$ endmembers should be used for a mixing model and beyond $6$ there is little added benefit. Based on these observations, multiple GSM models were trained with $N_v$ ranging from $3$ to $6$, $\lambda_e$ ranging from $0.001$ to $1.0$, and $\lambda_w$ ranging from $1$ to $1000$ in order to explore the GSM parameter space. From these, a final model was identified using the BIC, AIC, and reconstruction RMSE. The resulting GSM was then explored to examine extracted endmembers and map the evolution of the rhodamaine plume by using the abundances given by the latent space representation of each pixel in the HSI.



\section{Results}\label{sec:results}
\subsection{Linear Mixing}

The results of GSM training on the synthetic linear mixing dataset described in Sec~\ref{sec:experiments} are illustrated in Figure~\ref{fig:usgs-fits}. Three versions of NMF were trained corresponding to Euclidean ($\ell_2$), KL-Divergence, and $\ell_{2,1}$ cost functions. GSM models using both a regular simplex grid and a grid of points sampled using a uniform Dirichlet distribution (referred to as a \textit{big} GSM model) were trained to compare performance on linear mixing tasks.

\begin{figure}[H]
\includegraphics[width=\columnwidth]{results/usgs/fit-comparison.pdf}
\caption{Comparison of GSM against NMF on simulated linear mixing dataset using USGS spectra. \textbf{(a)} The mean spectral angle computed between extracted endmembers and original endmembers. \textbf{(b)} the mean RMSE computed between extracted endmembers and original endmembers. \textbf{(c)} The mean abundance RMSE computed between original abundance data for each endmember and extracted abundances. \textbf{(d)} The reconstruction RMSE which evaluates the quality of fit. All models realized similar values reflecting convergence of the models to the level of random noise introduced into the data.}
 \label{fig:usgs-fits}
\end{figure}  

The quality of endmember extraction is measured by the mean spectral angle and mean endmember RMSE. As Figure~\ref{fig:usgs-fits} indicates, both versions of the GSM outperformed their NMF counterparts. Additionally, we note that for all GSM models, even including $\text{SNR}=0$, all model weights $W_{dm}$ for $m>N_v$ corresponding to nonlinear mixing were driven identically to $0.0$. This confirms that for datasets with purely linear mixing and random noise, the GSM correctly fits a mixing model without introducing unnecessary complexity.

The quality of unmixing, i.e. abundance estimation, performed by each model was evaluated using the mean abundance RMSE. Here we again see that both versions of the GSM outperformed NMF. To justify that all models were fairly trained to convergence, the data reconstruction RMSE was also computed. This metric uses the trained mixing model to compute the error between the original dataset and the reconstructed spectra generated using the extracted endmembers and their abundances. For all GSM and NMF models, the data reconstruction RMSE converged to the level of random noise introduced into the data. These results reflect a fair comparison between NMF and GSM models.

In Figure~\ref{fig:usgs-endmembers}, we plot the extracted endmembers for the GSM model trained on the synthetic dataset with $\text{SNR}=20$.
\begin{figure}[H]
\begin{adjustwidth}{-\extralength}{-4.25cm}
\begin{center}
\includegraphics[width=1.1\columnwidth]{results/usgs/extracted-endmembers.png}
\end{center}
\end{adjustwidth}
\caption{Endmembers extracted by the GSM for the simulated linear mixing dataset with SNR$=20$. Dashed lines correspond to orignal endmember spectra from the USGS spectral database. Solid lines superimposed on the plot indicate the extracted endmember spectra. Colored bands are included around each spectrum corresponding to the spectral variability estimated by the GSM precision parameter $\beta$ where the band width is $2\sqrt{\beta^{-1}}$ corresponding to $2$ standard deviations.}
\label{fig:usgs-endmembers}
\end{figure}  
The extracted endmembers clearly fit the original source spectra while capturing local reflectance features. Additionally, the GSM precision parameter $\beta$ which is tuned during model training provides an assessment of spectral variability due to random noise. In Figure~\ref{fig:usgs-endmembers} this is indicated by the colored band centered around each extracted spectrum with a width of $2\sqrt{\beta^{-1}}$ corresponding to 2 standard deviations. The SNR of $20$ added to this example corresponds to zero-mean Gaussian noise with a standard deviation of $\sigma=0.0493$. After training the GSM found $\sqrt{\beta^{-1}}=0.0495$ which accurately captures the introduced noise. This ability to assess the spectral variability of extracted endmembers is a key advantage of the GSM.

\subsection{Nonlinear Mixing: Rhodamine Dye Plume}

For the dataset of real HSI spectra described in Section~\ref{sec:experiments}, 80 GSM models were trained to explore the GSM hyperparameter space. Model performance was compared using the BIC, AIC, and reconstruction RMSE with the results of the top $10$ performing models shown in Table~\ref{table:fit-comparison}. 
\begin{table}[H] 
\caption{GSM hyperparameter optimization for the real HSI dataset: Multiple GSM models were trained to explore the impact of model hyperparameters. $N_v$ values from $3$ to $6$ were explored as suggested by the PCA decomposition of the dataset. $\lambda_e$ was varied from $0.001$ to $1.0$ and $\lambda_w$ ranged from $1$ to $1000$. Here we report the top $10$ models ranked by increasing BIC. The AIC and reconstruction RMSE are also included for comparison.}
\label{table:fit-comparison}
\begin{tabularx}{\textwidth}{CCCCCC}
\toprule
\textbf{$N_v$}	& \textbf{$\lambda_e$}	& \textbf{$\lambda_w$} & \textbf{BIC} & \textbf{AIC} & \textbf{Reconstruction RMSE}\\
\midrule
$3$ & $0.01$    & $1.0$     & $-6.195\times10^7$   & $-6.269\times10^7$   & $0.000989$ \\
$3$	& $0.001$   & $1.0$     & $-6.194\times10^7$   & $-6.268\times10^7$   & $0.000989$ \\
$3$	& $0.1$     & $1.0$	    & $-6.192\times10^7$   & $-6.265\times10^7$   & $0.000991$ \\
$3$	& $1.0$     & $1.0$	    & $-6.186\times10^7$   & $-6.260\times10^7$   & $0.001190$ \\
$4$	& $1.0$     & $1.0$	    & $-6.181\times10^7$   & $-6.255\times10^7$   & $0.001002$ \\
$4$	& $0.1$     & $1.0$	    & $-6.175\times10^7$   & $-6.249\times10^7$   & $0.001008$ \\
$4$	& $0.01$    & $1.0$     & $-6.173\times10^7$   & $-6.247\times10^7$   & $0.001009$ \\ 
$4$	& $0.001$   & $1.0$     & $-6.173\times10^7$   & $-6.247\times10^7$   & $0.001009$ \\
$4$	& $0.1$     & $10.0$    & $-6.171\times10^7$   & $-6.245\times10^7$   & $0.001011$ \\
$3$	& $1.0$     & $10.0$    & $-6.166\times10^7$   & $-6.239\times10^7$   & $0.001014$ \\
\bottomrule
\end{tabularx}
\end{table}
As the table indicates, a nonlinear GSM with $N_v=3$, $\lambda_e=0.01$, and $\lambda_w=1.0$ achieved minimum BIC, AIC, and reconstruction RMSE values. The lower value of $\lambda_w$ identified from these models reflects the presence of nonlinear mixing effects in the HSI data. While the values of model weights $W_{dm}$ for $m>N_v$ corresponding to nonlinear mixing were $0$ for the synthetic dataset, here these weights obtained a small but non-negligible median value of $0.0012$. 

Reflectance spectra generated by the trained GSM corresponding to maximum abundance values for each endmember are plotted in Figure~\ref{fig:robot-team-endmembers}. Based on their signatures, these endmembers are identified with water, vegetation (including near shore, filamentous blue-green algae) and the rhodmaine dye plume.

\begin{figure}[H]
\begin{adjustwidth}{-\extralength}{0cm}
\centering
\includegraphics[width=1.25\columnwidth]{results/robot-team/extracted-endmembers.pdf}
\end{adjustwidth}
\caption{GSM applied to water spectra from real HSI dataset: \textbf{(a)} Spectra generated by the trained GSM for samples with maximum abundance for each endmember. Based on these spectral profiles, endmembers are identified with water, near-shore vegetation, and rhodamine dye. \textbf{(b)} A HSI segmented according to the relative abundance of each endmember. Each water pixel is colored by smoothing interpolating between red, green, and blue colors using the relative abundance estimated for rhodamine, vegetation, and water spectra. The rhodamine plume is clearly identifiable in the western portion of the HSI.}
\label{fig:robot-team-endmembers}
\end{figure}  

By assigning a unique color to each endmember, the spatial distribution of HSI spectra can be visualized by using estimated abundances to smoothly interpolate between endmember colors as shown in panel b of Figure~\ref{fig:robot-team-endmembers}. In other words, this method provides a fuzzy semantic segmentation of HSI pixels. Alternatively, each HSI pixel could be assigned a single class corresponding to the endmember with maximal abundance. In this way, the GSM can be used to visualize high-dimensional HSI spectra.

To showcase the ability of the GSM to identify spatially localized contaminant sources, abundance maps for each individual endmember were generated by using the trained GTM to unmix all water pixels the HSI. In Figure~\ref{fig:endmember-abundance-dist} these abundance maps are compared for each endmember. In particular, the abundance map for the water endmember covers a majority of the center of the pond and decreases near the edge of the rhodamine plume where the dye and water mix. Vegetation in shallow water near the shore and the rhodamine dye plume in the western half of the pond are also clearly identified by their abundance maps. Given a HSI pixel resolution of $0.1 \times 0.1$ $\text{m}^2$, the spatial extent of vegetation is estimated to be $378.6$ $\text{m}^2$ while the plume initially extends across $255.7$ $\text{m}^2$.

\newpage
\begin{figure}[H]
\begin{adjustwidth}{-\extralength}{0cm}
\centering
\includegraphics[width=\columnwidth]{results/robot-team/endmember-abundances.pdf}
\caption{Endmember abundance distributions: \textbf{(a)} The spatial distribution of abundance for the water class. This source dominates in the center of the pond and decreases towards the shore where vegetation begins to dominate the reflectance signal. The water endmember abundance is also observed to decreases near the edge of the rhodamine plume reflecting dye mixing and diffusion. \textbf{(b)} The spatial distribution of vegetation. This endmember includes filamentous blue-green algae observed to accumulate in shallow waters near the shore. \textbf{(c)} The rhodamine dye plume extent segmented from the HSI. The total area for near-shore vegetation and rhodamine are estimated to be $378.6$ $\text{m}^2$ and $255.7$ $\text{m}^2$, respectively.}
\label{fig:endmember-abundance-dist}
\end{adjustwidth}
\end{figure}  
\newpage


\begin{figure}[H]
\begin{adjustwidth}{-\extralength}{0cm}
\centering
\includegraphics[width=1.25\columnwidth]{results/robot-team/plume-evo.pdf}
\end{adjustwidth}
\caption{Rhodamine plume evolution: Using the trained GSM we can track the dispersion of the rhodamine dye plume between successive drone flights. \textbf{(a)} The initial plume distribution after release. Here the dye subsumes an area of $255.7$ $\text{m}^2$. \textbf{(b)} The same plume imaged 15 minutes later now extends across an area of $571.8$ $\text{m}^2$}
\label{fig:plume-evo}
\end{figure}  

Applying the trained GSM to unmix HSI from UAV flights provides an efficient way to map contaminant dispersion. Figure~\ref{fig:plume-evo} demonstrates this by mapping the growing extent of the rhodamine dye between successive UAV flights. In just $15$ minutes, the plume expands from an initial area of $255.7$ $\text{m}^2$ to over $571.8$ $\text{m}^2$ as the dye disperses into the surrounding water. 

\section{Discussion}\label{sec:disucssion}

- Comparison to NMF on linear mixing problem
    - note that there are many versions of NMF with different constraints/regularization. The purpose here was to show that GSM is at least \textit{comparable} to NMF for linear unmixing problems
    - Further note that both NMF and GSM do not rely on assumption of pure pixels (unlike VCA and other methods) 
    - GSM also allows estimation of endmember variability via the precision parameter $\beta$ i.e. nodes in data space follow Gaussian distribution with $\sigma = \sqrt{\beta^{-1}}$

- Nonlinear GSM for robot team 
    - discuss identification of rhodamine plume and easy abundance mapping via the embedding coordinate
    - discuss intrinsic dimensionality identification via abundance sparsity and BIC
    - Discuss identification of non-linearity \textit{strength} via regularization parameter and BIC.
    - reconstruction rmse can also be used to evaluate the relative performance for fixed model size (i.e. k and m) with varying regularization strength

- Limitations of the GSM
    - slow training due to computation of dinstance matrix between full dataset and embedding space
    - An incremental version of the EM procedure as described in \cite{gtm-developments} can speed this up by considering batches $X_b$ of $X$ and performing only a partial E-step by updating responsibilities corresponding to the data batch
    - However, M step still involves the "full" responsibility matrix. 
    - This can be addressed by taking an ensembling approach as described in \cite{parallel-gtm} where multiple GTM are trained on subsets of $X$ and the results are averaged. 
    - More generally, we may consider a mixture-of-GTMs as briefly desribed in \cite{gtm-orig} whereby the overall density is given as 
    \begin{equation}
        p(\mathbf{x}) = \sum_r P(r)p(\mathbf{x}\vert r)
    \end{equation}
    where there are $r$ individual GTM models with mixing coefficients $P(r)$ such that $\sum_r P(r) =1$

- Other potential extensions to GSM model
     - Adapt latent space  to allow for points whose barycentric coordinates sum to less than 1 in order to account for lighting effects (shouldn't matter for reflectance but would matter for radiance depending on incident light). This is the $\gamma$ factor that shows up in some of the mixing papers.
    - Augment the data-space distribution to allow for more complicated noise models than a fixed $\beta$ parameter. A diagonal covariance matrix would allow the model to infer seperate uncertainty criterion for each wavelength bin which is relevant for real world HSI which may have different noise levels for different wavelength bins.
    
- Discuss other applications of the GSM
    - Optimal route planning for in-situ data collection based on prize-collecting travelling salesman problem, i.e. construct a route which maximizies area traversed in GSM latent space (e.g. the n-simplex). 
    - Applications to source apportionment, e.g. linear GSM can be used for standard source apportionment problems (no reactions) and nonlinear GSM may be useful for modeling scenarios with additional complications due to reactions and during transport from source to receptor.

\section{Conclusions}\label{sec:conclusions}


\section{Patents}

\textcolor{red}{UPDATE ME!}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\vspace{6pt} 


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\authorcontributions{Methodology, J.W.; conceptualization, J.W.; software, J.W.; validation, J.W.; formal analysis, J.W.; investigation J.W.; resources, D.J.L.; writing---original draft preparation, J.W.; writing---review and editing, J.W. and D.J.L.; visualization, J.W.; supervision, D.J.L.; project administration, D.J.L.; funding acquisition, D.J.L. All authors have read and agreed to the published version of the manuscript.
}


\funding{\textcolor{red}{Please add: ``This research received no external funding'' or ``This research was funded by NAME OF FUNDER grant number XXX.'' and  and ``The APC was funded by XXX''. Check carefully that the details given are accurate and use the standard spelling of funding agency names at \url{https://search.crossref.org/funding}, any errors may affect your future funding.}}


\institutionalreview{Not applicable.}

\informedconsent{Not applicable.}


\dataavailability{\textcolor{red}{We encourage all authors of articles published in MDPI journals to share their research data. In this section, please provide details regarding where data supporting reported results can be found, including links to publicly archived datasets analyzed or generated during the study. Where no new data were created, or where data is unavailable due to privacy or ethical restrictions, a statement is still required. Suggested Data Availability Statements are available in section ``MDPI Research Data Policies'' at \url{https://www.mdpi.com/ethics}.}}


\acknowledgments{\textcolor{red}{In this section you can acknowledge any support given which is not covered by the author contribution or funding sections. This may include administrative and technical support, or donations in kind (e.g., materials used for experiments).}}

\conflictsofinterest{The authors declare no conflicts of interest.}
 

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Optional

%% Only for journal Encyclopedia
%\entrylink{The Link to this entry published on the encyclopedia platform.}

\abbreviations{Abbreviations}{
The following abbreviations are used in this manuscript:\\

\noindent
\begin{tabular}{@{}ll}
PRISMA & Hyperspectral Precuror of the Application Mission \\
EnMAP & Environmental Mapping and Analysis Program \\
PACE & Plankton, Aerosol, Cloud, ocean Ecosystem \\
CHIME & Copernicus Hyperspectral Imaging Mission for the Environment \\
NIR & Near Infrared \\
SWIR & Short-wave Infrared \\
UAV & Unmanned Aerial Vehicle \\
HSI & Hyperspectral Image \\
NDVI & Normalized Difference Vegetation Index \\
VCA & Vertex Component Analysis\\
PPI & Pixel Purity Index \\ 
LMM & Linear Mixing Model \\
NMF & Non-negative Matrix Factorization \\ 
SOM & Self-Organizing Map \\ 
GTM & Generative Topographic Mapping  \\ 
EM & Expectation-Maximization \\
GSM & Generative Simplex Mapping \\ 
RBF & Radial Basis Function \\
RMSE & Root Mean Square Error \\ 
PCA & Principal Component Analysis
\end{tabular}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Optional
\appendixtitles{no} % Leave argument "no" if all appendix headings stay EMPTY (then no dot is printed after "Appendix A"). If the appendix sections contain a heading then change the argument to "yes".


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{adjustwidth}{-\extralength}{0cm}
%\printendnotes[custom] % Un-comment to print a list of endnotes

\reftitle{References}

\bibliography{references.bib}
\PublishersNote{}

\end{adjustwidth}
\end{document}

